File: java/vansh_jain/trees/PathSum.java
static int solve(TreeNode root, int targetSum){
    //CODE HERE 
        if(pathSum(root,targetSum))
            return 1;
        return 0;
    
    
    }
    
    static boolean pathSum(TreeNode node, int sum){
        if(sum==0)
            return true;
        if(node==null)
            return false;
        return pathSum(node.left,sum-node.val) || pathSum(node.right, sum-node.val);
    }

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/pathSum.py
def solve(root, targetSum):
    ans = 0
    subSum = targetSum - root.val
 
    # If we reach a leaf node and sum becomes 0, then
    # return True
    if(subSum == 0 and root.left == None and root.right == None):
        return 1
 
    # Otherwise check both subtrees
    if root.left is not None:
        ans = ans or solve(root.left, subSum)
    if root.right is not None:
        ans = ans or solve(root.right, subSum)
 
    return ans